analyzers/src/SonarAnalyzer.Common/Rules/RouteTemplateShouldNotStartWithSlashBase.cs

```diff
@@ -74,18 +74,14 @@ private void ReportIssues(SonarSymbolReportingContext context, INamedTypeSymbol
         var issueMessage = controllerSymbol.GetAttributes().Any(x => x.AttributeClass.IsAny(KnownType.RouteAttributes) || x.AttributeClass.Is(KnownType.System_Web_Mvc_RoutePrefixAttribute))
             ? MessageOnlyActions
             : MessageActionsAndController;
-        var attributeLocations = actions.SelectMany(x => x.RouteParameters).Select(x => x.Key);
 
-        if (ControllerDeclarationSyntax(controllerSymbol) is { } controllerSyntax
-            && Language.Syntax.NodeIdentifier(controllerSyntax) is { } identifier)
+        var secondaryLocations = actions.SelectMany(x => x.RouteParameters).Select(x => x.Key);
```

Comment:
I think something is off. We push the template arguments per method symbol but we never check the symbol when we get the parameters.

This makes me think that either `IMethodSymbol Action` is not needed on `ControllerActionInfo` or we have an issue.

Do I miss something?

Reply:
I was using it before to to check the class that a method belongs to - but I don't do this anymore.
I'll remove it!

Reply:
You can simplify the code by removing the `ControllerActionInfo` record and passing the dictionary around.

Reply:
Done